airflow/api_connexion/security.py

```diff
@@ -48,10 +61,194 @@ def requires_access_decorator(func: T):
+    :param args: the arguments of ``func``
+    :param kwargs: the keyword arguments ``func``
+
+    :meta private:
+    """
+    check_authentication()
+    if is_authorized_callback():
+        return func(*args, **kwargs)
+    raise PermissionDenied()
+
+
+def requires_authentication(func: T):
+    """Decorator for functions that require authentication."""
+
+    @wraps(func)
+    def decorated(*args, **kwargs):
+        check_authentication()
+        return func(*args, **kwargs)
+
+    return cast(T, decorated)
+
+
+def requires_access_configuration(method: ResourceMethod) -> Callable[[T], T]:
+    def requires_access_decorator(func: T):
+        @wraps(func)
+        def decorated(*args, **kwargs):
+            section: str | None = kwargs.get("section")
+            return _requires_access(
+                is_authorized_callback=lambda: get_auth_manager().is_authorized_configuration(
+                    method=method, details=ConfigurationDetails(section=section)
+                ),
+                func=func,
+                args=args,
+                kwargs=kwargs,
+            )
+
+        return cast(T, decorated)
+
+    return requires_access_decorator
+
+
+def requires_access_connection(method: ResourceMethod) -> Callable[[T], T]:
+    def requires_access_decorator(func: T):
+        @wraps(func)
+        def decorated(*args, **kwargs):
+            connection_id: str | None = kwargs.get("connection_id")
+            return _requires_access(
+                is_authorized_callback=lambda: get_auth_manager().is_authorized_connection(
+                    method=method, details=ConnectionDetails(conn_id=connection_id)
+                ),
+                func=func,
+                args=args,
+                kwargs=kwargs,
+            )
+
+        return cast(T, decorated)
+
+    return requires_access_decorator
+
+
+def requires_access_dag(
+    method: ResourceMethod, access_entity: DagAccessEntity | None = None
+) -> Callable[[T], T]:
+    appbuilder = get_airflow_app().appbuilder
+
+    def _is_authorized_callback(dag_id: str):
+        def callback():
+            access = get_auth_manager().is_authorized_dag(
+                method=method,
+                access_entity=access_entity,
+                details=DagDetails(id=dag_id),
+            )
+
+            # ``access`` means here:
+            # - if a DAG id is provided (``dag_id`` not None): is the user authorized to access this DAG
+            # - if no DAG id is provided: is the user authorized to access all DAGs
+            if dag_id or access:
+                return access
+
+            # No DAG id is provided and the user is not authorized to access all DAGs
+            # If method is "GET", return whether the user has read access to any DAGs
+            # If method is "PUT", return whether the user has edit access to any DAGs
+            return (method == "GET" and any(appbuilder.sm.get_readable_dag_ids())) or (
```

Comment:
Is there a readson why this check is here and not in the FAB implementation ? I believe FAB manager should be able to check it on its own in the `is_authorized_dag` method and there is no need to do this check in the decorator? Or maybe I am missing something?

Reply:
I dont think it should be in the auth manager. This is a behavior common to all auth managers. When a function is decorated with `@requires_access_dag("GET")` with no `dag_id` provided, it means "does the user has access to read at least one DAG". For instance, the index page (where DAGs are listed) is decorated that way.

So saying that, in case of `GET` (works similar with `PUT`), we want to check whether the user has access to at least one DAG is agnostic to auth managers, am I wrong? The auth manager plays a role to return the list of DAG accessible by the user but the overall logic "does the user has access to read at least one DAG" is generic to Airflow to me.

WDYT?

Reply:
Fine for me. But we should not use "appbuilder.sm" here -it should be "get_readable_dag_ids" method from "get_auth_manager".

Reply:
Also that plays nice with actually having "get_readable_dag_ids" as separate method (and default implementation involving iterating over all dags and individually checking if it has access - see the https://github.com/apache/airflow/pull/34317#discussion_r1355994191

Reply:
Done. I agree. I like it that way and open for optimization on the auth managers side :+1: